Objective-C/CBLProperties.mm

```diff
@@ -146,101 +144,346 @@ - (BOOL) containsObjectForKey: (NSString*)key {
 
 
 - (nullable id) objectForKeyedSubscript: (NSString*)key {
-    if (_properties)
-        return _properties[key];
-    else
-        return [self fleeceValueToObject: [self fleeceValueForKey: key]];
+    id obj = _properties ? _properties[key] : nil;
+    if (!obj) {
```

Comment:
This method doesn't handle removed properties. If a property exists in `_root`, and then I set its value to `nil`, then this method _should_ return the new value of `nil`. But this code will see that there's no value in `_properties`, and fall back to reading the value from Fleece.
This line should also be checking whether `key` is in `_changedNames`, and if it is, skipping the following block. I.e. `if (!obj && ![_changedNames containsObject: key])`.
I don't know if you've looked at the property-handling code of CBLModel, but it does this.